diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index 371a6e54c..747c0b45c 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -2623,7 +2623,7 @@ SDL_SetWindowGammaRamp(SDL_Window * window, const Uint16 * red,
     CHECK_WINDOW_MAGIC(window, -1);
 
     if (!_this->SetWindowGammaRamp) {
-        return SDL_Unsupported();
+        return 0/*SDL_Unsupported()*/;
     }
 
     if (!window->gamma) {
