diff --git a/components/misc/strings/format.hpp b/components/miscstrings/format.hpp
index fe4be0000..b89185919 100644
--- a/components/misc/strings/format.hpp
+++ b/components/misc/strings/format.hpp
@@ -9,6 +9,7 @@
 #include <string_view>
 #include <system_error>
 #include <vector>
+#include <boost/format.hpp>
 
 namespace Misc
 {
@@ -55,12 +55,14 @@
            if (size < 0)
                throw std::system_error(errno, std::generic_category(), "Failed to compute resulting string size");
+
+        /*auto size = std::snprintf(nullptr, 0, fmt, argument(args) ...);
         // Note: sprintf also writes a trailing null character. We should remove it.
         std::string ret(size+1, '\0');
         std::sprintf(&ret[0], fmt, argument(args) ...);
         ret.erase(size);
 
-        return ret;
+        return ret;*/
+        return boost::str((boost::format(fmt) % ... % args));
         
      }
  }
