diff --git a/apps/openmw/mwstate/statemanagerimp.cpp b/apps/openmw/mwstate/statemanagerimp.cpp
index 9f572fde9..bdad92a2d 100644
--- a/apps/openmw/mwstate/statemanagerimp.cpp
+++ b/apps/openmw/mwstate/statemanagerimp.cpp
@@ -14,6 +14,7 @@
 #include <components/settings/settings.hpp>
 
 #include <osg/Image>
+#include <osg/Hint>
 
 #include <osgDB/Registry>
 
@@ -309,6 +310,8 @@ void MWState::StateManager::saveGame (const std::string& description, const Slot
         Settings::Manager::setString ("character", "Saves",
             slot->mPath.parent_path().filename().string());
 
+        osg::Hint::GL4ES_SavePSA();
+
         const auto finish = std::chrono::steady_clock::now();
 
         Log(Debug::Info) << '\'' << description << "' is saved in "
